apps/glusterfs/volume_entry.go

```diff
@@ -154,6 +155,34 @@ func NewVolumeEntryFromId(tx *bolt.Tx, id string) (*VolumeEntry, error) {
 	return entry, nil
 }
 
+func NewVolumeEntryFromClone(v *VolumeEntry, name string) *VolumeEntry {
+	entry := NewVolumeEntry()
+
+	entry.Info.Id = utils.GenUUID()
+	if name == "" {
+		entry.Info.Name = "vol_" + entry.Info.Id
+	} else {
+		entry.Info.Name = name
+	}
+
+	entry.GlusterVolumeOptions = v.GlusterVolumeOptions
```

Comment:
Does a cloned volume get the same volume options too? If yes then the code is fine. Can we add a comment for the same? If it does not then let us make this empty.

Reply:
Yes, I think that clone should copy the options. Any reason not to?